discussions/prog_repo.md

```diff
@@ -0,0 +1,192 @@
+- The L3AFD daemon, which is a major component of the L3AF Project.
+
+We consider L3AF to be an entire ecosystem that aims to provide eBPF Programs as a service.
+We've otherwise phrased this as "complete lifecycle management of eBPF programs".
+This definition includes:
+
+- The L3AFD daemon, an orchestrator that provides APIs to launch and manage eBPF programs on a node
+- The L3AF eBPF Package Repository
+- Programs within the eBPF Package Repository
+
+Note, however, that we have no intention of limiting the creation of other public or private
+eBPF Package repositories.
+
+We recognize that it may make sense to migrate the eBPF Package Repository
+(eBPF Programs within the repository) out of the L3AF Project and into its own
+Linux Foundation project in the future. Doing this initially may not make sense (due to L3AF-specific
+eBPF program chaining implementation for networking programs), but as L3AF and other projects mature,
+a platform-agnostic repository could be useful for multiple projects.
+
+# What should an initial version look like
+
+This section examines simple ways to create a location where eBPF Programs can be uploaded
+and made available for others to download.
+
+## A GitHub repository may be sufficient
+
+We would like to leverage GitHub:
+
+- A new GitHub repository will be created to store eBPF program source code
+- All submissions will be manually reviewed by the L3AF team
+- Once approved, programs will be published in the eBPF Package Repository
+
+Another important thing to note is that, initially, code submissions will need
+to conform to L3AF's eBPF program chaining mechanics.
+
+## To Build or Not to Build
+
+Because of the overhead and support requirements of a full build system, it may not be
+feasible to build the eBPF program source code in the initial version. However, we believe
+this would greatly hinder adoption from both contributors and users. Our proposal, therefore,
+is to have the repository’s initial version include scripts (e.g. Dockerfile for build system images),
+and steps to build eBPF Programs locally.
+
+## eBPF Package Repository of the future
+
+Future versions could build on top of the foundation laid by the initial version.
+
+# What should a future version look like
+
+This section examines simple ways to create a location where eBPF Programs from trusted parties
+can be uploaded and made available for others to download. This can also enable users to provide
+rating, reviews, and tags to the packages.
+
+### Build Process
+
+Contributed eBPF programs can be built on common images for Linux and other platforms.
+Contributors are expected to build their code using the most recent image used by our build system.
+Contributors could have a choice to build on Linux only or other platforms.
+
+Build artifacts (i.e., eBPF program bytecode and user space binaries) could be stored
+in a public file storage repository. Users could download the artifacts directly
```

Comment:
Aren't those inside the eBPF Packages uploaded?

Reply:
@dthaler Yes, here artifacts mean nothing but eBPF packages.